### PR TITLE
feat: transfer docker images from public registries to aws private ECR

### PR DESCRIPTION
**Pull request by nitesh-balla**

### CODE REVIEW

File: dependencies/code_builder/buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+version: 0.2
+
+phases:
+  pre_build:
+    commands:
+      - echo "Logging in to the docker"
```
Comment (inventvenkat) [inline dependencies/code_builder/buildspec.yml:6]: Can we remove this echo?

File: dependencies/code_builder/buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  build:
+    commands:
+       - echo "Building and pushing Docker images"
```
Comment (inventvenkat) [inline dependencies/code_builder/buildspec.yml:9]: the message in the echo is different from the operations followed. Please use the proper comments

File: lib/aws/eks.ts [hunk 3]
```diff
@@ -720,3 +726,106 @@ class KmsSecrets {
...
+
+    const ecrRole = new iam.Role(scope, "ECRRole", {
+      assumedBy: new iam.ServicePrincipal("codebuild.amazonaws.com"),
+    });
+
+    const ecrPolicy = new iam.PolicyDocument({
+      statements: [
+        new iam.PolicyStatement({
+          actions: [
+            "ecr:*",
```
Comment (inventvenkat) [inline lib/aws/eks.ts:745]: Can you give the exact permissions, instead of all

**Review by inventvenkat — CHANGES_REQUESTED** [review]

File: dependencies/code_builder/buildspec.yml (hunk 1 above)
Comment (nitesh-balla) [reply dependencies/code_builder/buildspec.yml:6]: Yes

File: dependencies/code_builder/buildspec.yml (hunk 2 above)
Comment (nitesh-balla) [reply dependencies/code_builder/buildspec.yml:9]: It's not required, we ccan remove it

**Review by inventvenkat — APPROVED** [review]

**Pull request merged**